### ISSUE TITLE
Resnet-101 trainable despite setting is_training flag to False

### ISSUE THREAD

**Issue by akku1506** [tensorflow/models#5261]
Hello,

I was extracting the pre-trained features of Resnet-101 and then finetune a small net from the features I get from Resnet-101.

I followed the API and wrote the below lines of code to extract-2048 dimesnioanll features:

 with slim.arg_scope(resnet_v1.resnet_arg_scope()):
            _, end_points = resnet_v1.resnet_v1_101(images, None, is_training=False)

But when, i do a simple check of getting all the trainable variables in my code **by calling tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES), I found all the parameters of Resnet-101 trainable**.

Further, I also inspected the code and found that maybe there should be a global scope defined for slim.conv2d, slim.batch_norm which should be set to False in resnet_utils, if is_training=False to extract pre-trained features. 

This also means that the entire network is getting trained, which I don't expect if I am setting is_training=False.

What is the top-level directory of the model you are using: models/research/slim/nets
Have I written custom code: No
OS Platform and Distribution :Ubuntu 16.04 LTS
TensorFlow installed from: Anaconda and this library from (https://github.com/tensorflow/models.git)
TensorFlow version: 1.9
Reproduce the issue by: 

with slim.arg_scope(resnet_v1.resnet_arg_scope()):
            _, end_points = resnet_v1.resnet_v1_101(images, None, is_training=False)
and then printing all trainable variable name from tf.get_collection(tf.GraphKeys.TRAINABLE_VARIABLES)

**Comment by akku1506** [comment]
Updated the issue! Its a bus in the code of models/slim/nets/resnet_v1.py

**Comment by pkulzc** [comment]
is_training has a different meaning from what you expect. See definition [here](https://github.com/tensorflow/models/blob/master/research/slim/nets/resnet_v1.py#L182). It's used for freezing batch norm instead of vars. 

variables are trainable doesn't mean they will be trained. If no loss were computed and no optimized step were run, they will not be updated.

If you want to freeze some vars and train the rest, use this [config](https://github.com/tensorflow/models/blob/master/research/object_detection/protos/train.proto#L71) instead.

**Comment by tensorflowbutler** [comment]
Hi There,
 We are checking to see if you still need help on this, as this seems to be considerably old issue. Please update this issue with the latest information, code snippet to reproduce your issue and error you are seeing.
 If we don't hear from you in the next 7 days, this issue will be closed automatically. If you don't need help on this issue any more, please consider closing this.
